src/sources/internal_metrics.rs

```diff
@@ -13,16 +13,30 @@ use futures::{
 use futures01::Sink;
 use serde::{Deserialize, Serialize};
 use std::time::Duration;
-use tokio::{select, time::interval};
+use tokio::select;
 
 #[derive(Deserialize, Serialize, Debug, Clone, Default)]
-pub struct InternalMetricsConfig {}
+pub struct InternalMetricsConfig {
+    #[serde(default = "default_scrape_interval_secs")]
+    scrape_interval_secs: u64,
+}
+
+pub fn default_scrape_interval_secs() -> u64 {
+    2
+}
 
 inventory::submit! {
     SourceDescription::new::<InternalMetricsConfig>("internal_metrics")
 }
 
-impl_generate_config_from_default!(InternalMetricsConfig);
+impl GenerateConfig for InternalMetricsConfig {
```

Comment:
I think we could implement `Default` here instead. That would have the advantage of also being able to use `InternalMetricsConfig::default()` in tests and other places.